lib/puppet/indirector/yaml.rb

```diff
@@ -19,17 +19,20 @@ def find(request)
   def save(request)
     raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
 
-    file = path(request.key)
+    files = path(request.key)
```

Comment:
If it's dead code, my thought was we should remove it and throw exceptions. However, since the default setting is a single path, it's relatively safe to leave it. I'd probably accept comments on each of the other methods mentioning they don't currently support multiple paths, but that the code is essentially unused; if anyone ever uses it they'll need to extend in the ways you mention.